### PR TITLE
Bug #1923 search results give the wrong count

### PR DESCRIPTION
**Pull request by bstutes**
Made simple change agreed upon in the discussion thread for this bug.  Also found the <label...> was not properly associated with the <select....> tag to specify the sort order for returning projects found in a search.  Added "id=" attribute to properly associate the label.

Also cleaned up comments that document all changes made.  That's the reason for commits 85c2b0c, 5682ad0 and 87b1cc7.

### CODE REVIEW

File: warehouse/templates/search/results.html [hunk 1]
```diff
@@ -35,9 +35,6 @@ <h3 class="package-snippet__title">
```
Comment (di) [inline warehouse/templates/search/results.html]: Looks like the line above this one (`% set freq = best_guess.freq %}`) can be removed as well.

Comment (bstutes) [inline warehouse/templates/search/results.html]: I'll make the additional change and create a new pull request.

Comment (di) [inline warehouse/templates/search/results.html]: @bstutes Making a new PR shouldn't be necessary, you should be able to just push a new commit to your existing branch.

File: warehouse/templates/search/results.html [hunk 2]
```diff
@@ -143,7 +139,7 @@ <h2 class="no-top-padding">Filter Projects</h2>
           <div>
             <input id="search" type="hidden" name="q" value="{{ term }}">
             <label for="order">Order by &nbsp;</label>
-            <select class="-js-form-submit-trigger" name="o">
+            <select class="-js-form-submit-trigger" id="order" name="o">
```
Comment (dstufft) [inline warehouse/templates/search/results.html]: What's the purpose of this change?

Comment (di) [inline warehouse/templates/search/results.html]: It correctly makes the label correspond to the form element via the `for` attribute.

Comment (dstufft) [inline warehouse/templates/search/results.html]: Ahh, I missed that. Makes sense.
